### ISSUE TITLE
Unable to compile old version of hamsterdb

### ISSUE THREAD

**Issue by SkyToGround** [cruppstahl/upscaledb#81]
I have a database from 2012 which I am unable to open with the latast version of upscaledb. The older releases of hamsterdb downloaded from here does not have a configure-file and trying to generate one using autoconf yields the following errors:

```
configure.in:13: error: possibly undefined macro: AM_INIT_AUTOMAKE
      If this token and others are legitimate, please use m4_pattern_allow.
      See the Autoconf documentation.
configure.in:15: error: possibly undefined macro: AM_CONFIG_HEADER
configure.in:20: error: possibly undefined macro: AC_LIBTOOL_WIN32_DLL
configure.in:21: error: possibly undefined macro: AC_PROG_LIBTOOL
configure.in:102: error: possibly undefined macro: AM_CONDITIONAL
```

I have tried a few different suggestions from around the web with no success. I am open to any suggestions on how to extract the data from the old database.

**Comment by cruppstahl** [comment]
Which version do you need? Would a win32 library also work for you? I might find a precompiled version lying around somewhere...

**Comment by SkyToGround** [comment]
The database file was originally created using some 1.x version of hamsterdb I believe. I am also pretty sure that I was able access the data using some early 2.x version but I have no idea at what version backward compatibility was broken.

I have tried several different hamsterdb versions here but none that I have tried have a configure-file or a working configure.in-file.

As I do most of my development on MacOSX and some on Linux I would prefer a solution using one of those two operating systems. Using win32 would be an option if no other exist but I would like to avoid it as I have no experience in Windows based development environments.

**Comment by cruppstahl** [comment]
I suspect that your automake/autoconf is "too new" for the old hamsterdb configure script.

I think there might be three options:
1. try installing older versions of automake/autoconf
2. hack configure.in, then run bootstrap.sh (https://github.com/cruppstahl/upscaledb/blob/master/bootstrap.sh) to generate configure.in, and see if it works
3. use a Win32 build to extract the data, i.e. with "ham_dump" (which is part of the precompiled binaries), then i.e. use a perl script to further process the data

I think I would start with option 2.

BTW: if you open the file with upscaledb and set a breakpoint in upscaledb::LocalEnv::do_open (https://github.com/cruppstahl/upscaledb/blob/master/src/4env/env_local.cc#L311), then you can see in the debugger which version was used to create the file.

**Comment by SkyToGround** [comment]
Using an older version of autoconf did not work. I did not try to edit the `configure.in`-file as I am very unfamiliar with autoconf.

I found some old version 2.1.1 Win32 binaries on the internet with the `ham_dump.exe` application which worked with the database file (apparently v. 1.1.1). With the raw data extracted I used a python script to piece it together again.

For others who encounters the problem I had: remember to set "-maxrec" to a value larger than the largest record size in the database when using `ham_dump.exe`.
